aries_cloudagent/wallet_handler/handler.py

```diff
@@ -255,6 +302,16 @@ async def add_connection(self, connection_id: str, wallet_id: str):
         """
         self._connections[connection_id] = wallet_id
 
+    async def add_key(self, key: str, wallet_id: str):
+        """
+        Add a mapping between the given connection and wallet.
+
+        Args:
+            connection_id: Indentifier of the new connection.
+            wallet_id: Identifier of the wallet the connection belongs to.
+        """
+        self._handled_keys[key] = wallet_id
+
```

Comment:
Each verkey used  for a connection needs to be stored and mapped to the wallet to which the connection belongs.